kip-0039.md

```diff
@@ -0,0 +1,134 @@
+---
+KIP: '0039'
+Title: kadena_getNetwork_v1 API
+Author: Danillo Felixdaal and Bart Huijgen
+Status: Draft
+Type: Standard
+Category: Interface
+Created: 2025-02-17
+---
+
+# kadena_getNetwork_v1 API
+
+## Abstract
+
+This KIP standardizes the `kadena_getNetwork_v1` method for retrieving information about the currently active network from a Kadena wallet. The method provides consistent network details to dApps, ensuring they can correctly adjust to the appropriate blockchain environment.
+
+## Motivation
+
+A uniform approach to obtaining network information simplifies dApp development by ensuring that network details are provided in a consistent format across different wallet implementations. This standardization will:
+
+- Provide reliable network details (such as network name and networkId) to dApps.
+- Simplify integration and reduce potential mismatches between wallet and dApp network settings.
+- Enhance the overall user experience by ensuring clear network context.
+
+## Specification
+
+### Method Definition
+
+**Method Name:** `kadena_getNetwork_v1`
+
+### Request Parameters
+
+```typescript
+interface GetNetworkRequest {
+  // No parameters are required for this method.
+}
+```
+
+### Response
+
+The method returns a NetworkInfo object containing details about the active network:
+
+```typescript
+interface NetworkInfo {
+  name: string // The human-readable name of the network.
+  networkId: string // The unique identifier for the network (e.g., "mainnet01").
+  url?: string // Optional: The network endpoint URL.
```

Comment:
Specify what this endpoint represents. Whether it should be just `api.chainweb.com` or the full path to where the commands can be sent eg.: `https://api.chainweb.com/chainweb/0.0/mainnet01/chain/1/pact/api/v1`.
Should this contain placeholders for chainId?

Reply:
I would say consistency is key so I agree.